File: content/examples/other/leetcode/problems/counting-bits/example-1/Solution.java
import java.util.Arrays;

public class Solution {
  static int[] countBitsNaive(int n) {
    int[] out = new int[n + 1];
    for (int i = 0; i <= n; i += 1) {
      int count = 0;
      for (int b = 0; b < 32; b += 1) count += (i >>> b) & 1;
      out[i] = count;
    }
    return out;
  }

  public static void main(String[] args) {
    System.out.println(Arrays.toString(countBitsNaive(5)));
  }
}

File: content/examples/other/leetcode/problems/counting-bits/example-1/solution.js
function countBitsNaive(n) {
  const out = new Array(n + 1).fill(0);
  for (let i = 0; i <= n; i += 1) {
    let x = i >>> 0;
    let count = 0;
    for (let b = 0; b < 32; b += 1) count += (x >>> b) & 1;
    out[i] = count;
  }
  return out;
}

if (require.main === module) {
  console.log(countBitsNaive(5));
}

module.exports = { countBitsNaive };
